src/software/gui/robot_diagnostics/widgets/robot_diagnostics_gui.h

```diff
@@ -62,6 +76,13 @@ class RobotDiagnosticsGUI : public QMainWindow
                                       DirectVelocityMode direct_velocity_mode);
     void onDribblerPowerChanged(double dribbler_power);
 
+    /**
+     * Sends a primitive set msg to the primitive_buffer
+     *
+     * @param primitive_msg The msg to send
+     */
+    void sendPrimitiveSet(std::unique_ptr<TbotsProto::Primitive> primitive_msg);
```

Comment:
sending is semantically because it sounds like we're sending something to the robots. Can we change to `pushPrimitiveSetToBuffer` or maybe `recordPrimitiveSet`